layouts/jobs/list.html

```diff
@@ -310,49 +360,64 @@ <h5 class="modal-title" id="descriptionModalLabel${ind}">${job.title}</h5>
         var job_type = $('input:checkbox[name="type"]:checked').map(function () {return this.value}).get();
         var employer = $('input:checkbox[name="employer"]:checked').map(function () {return this.value}).get();
     
+        // Cache DOM elements for better performance
+        var $jobsContainer = $('div#jobs-container');
+        var $jobs = $jobsContainer.find('div.job');
+        var $noFilteredJobs = $('div#no-filtered-jobs');
+        
         // Hide all displayed jobs and message
-        $('div#jobs-container div.job').hide();
-        $('div#no-filtered-jobs').hide();
+        $jobs.hide();
+        $noFilteredJobs.hide();
     
-        // Show relavant jobs based on selected criteria
+        // Show relevant jobs based on selected criteria
         if (!job_type.length && !employer.length) {
-          $('div#jobs-container div.job').show();
+          $jobs.show();
         } else {
-          job_type.forEach(function(jtype){
-            $(`div#jobs-container div[data-job-type*=${jtype}]`).each(function(){
-              if (!employer.length) {
-                this.style.display = '';
-              } else if (employer.includes(this.getAttribute('data-employer'))){
-                this.style.display = '';
-              }
-            });
-          });
-          employer.forEach(function(emp){
-            if (!job_type.length) {
-              $(`div#jobs-container div[data-employer*=${emp}]`).show();
+          // Use more efficient filtering
+          $jobs.each(function(){
+            var $job = $(this);
+            var jobType = $job.data('job-type');
+            var jobEmployer = $job.data('employer');
+            
+            var showJob = false;
+            
+            if (!job_type.length && !employer.length) {
+              showJob = true;
+            } else if (job_type.length && employer.length) {
+              // Both filters active
+              showJob = job_type.some(type => jobType.toLowerCase() === type.toLowerCase()) && 
```

Comment:
`jobType` will not be a list anymore, so maybe can directly use `.includes` like for `employer` ?